documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,462 @@
+We deploy the Kafka cluster configured to use one of the pre-configured realms.
+We connect to the Keycloak Admin Console and demonstrate the sections of the user interface where authorizations rules are configured.
+We use Kafka CLI client tools with different personal and service accounts to demonstrate limited access based on the permissions granted to different accounts.
+
+
+== Token Based Authorization with Keycloak Authorization Services
+
+Once the Kafka Broker has obtained an access token by using `oauth` authentication, it is possible to use centrally managed authorization rules to enforce access restrictions onto Kafka Clients.
+For this, Strimzi Kafka Operator comes with `keycloak` authorization which uses Keycloak Authorization Services to centrally manage permissions.
+
+When using `keycloak` authorization a custom authorizer is configured on the Kafka Broker that uses Authorization Services REST endpoints available on Keycloak, which provide a list of granted permissions on resources for authenticated users.
+The list of grants (permissions) is fetched as the first action after an authenticated session is established by the Kafka client, and then regularly refreshed in the background.
+Grants are cached and enforced locally on the Kafka Broker for each user session to provide fast authorization decisions. Because they are refreshed, any changes to the grants on the Keycloak server are detected and enforced.
+
+
+== Starting Up the Pods
+
+For this example we're using the kubernetes deployment scripts available at https://github.com/strimzi/strimzi-kafka-oauth/tree/master/examples/kubernetes
+
+[source,yaml]
+----
+export ROOT=https://raw.githubusercontent.com/strimzi/strimzi-kafka-oauth/master
+----
+
+.Deploy the Postgres database for Keycloak
+
+[source,yaml]
+----
+kubectl apply -f $ROOT/examples/kubernetes/postgres-pvc.yaml
+kubectl apply -f $ROOT/examples/kubernetes/postgres.yaml
+----
+
+.Deploy the Keycloak server
+
+[source,yaml]
+----
+kubectl apply -f $ROOT/examples/kubernetes/keycloak-realms-configmap.yaml
+kubectl apply -f $ROOT/examples/kubernetes/keycloak-postgres.yaml
+----
+
+If your default namespace is not `myproject` (for example, `default`), use the following to deploy Keycloak:
+
+[source,yaml]
+----
+curl -s $ROOT/examples/kubernetes/keycloak-postgres.yaml | sed -e "s#myproject#default#" | kubectl apply -f -
+----
+
+.Deploy the minimal Kafka cluster
+
+Here we assume that the Strimzi Cluster Operator has already been installed on your Kubernetes cluster.
+
+[source,yaml]
+----
+kubectl apply -f $ROOT/examples/kubernetes/kafka-oauth-single-authz.yaml
+----
+
+
+== Using the Keycloak Admin Console to configure authorization
+
+You can login to the Keycloak Admin Console by creating a tunnel to the keycloak pod:
+
+[source,yaml]
+----
+kubectl port-forward keycloak 8080
+----
+
+Use your browser to connect to `http://localhost:8080/auth/admin` and use `admin` as username and password.
```

Comment:
```suggestion
Use your browser to connect to `\http://localhost:8080/auth/admin` and use `admin` as username and password.
```

Reply:
I suppose this backslash becomes invisible when rendered but prevents some unwanted rendering that would otherwise occur?